spec/tbd_tests_spec.rb

```diff
@@ -2718,4 +2718,100 @@
       end
     end
   end
+
+  it "can generate multiple KIVA exposed perimeters (midrise apts - variant)" do
+    translator = OpenStudio::OSVersion::VersionTranslator.new
+    path = OpenStudio::Path.new(File.dirname(__FILE__) + "/files/midrise_KIVA.osm")
+    os_model = translator.loadModel(path)
+    expect(os_model.empty?).to be(false)
+    os_model = os_model.get
+
+    # Reset all ground-facing surfaces as "foundations".
+    os_model.getSurfaces.each do |s|
+      next unless s.outsideBoundaryCondition.downcase == "ground"
+      s.setOutsideBoundaryCondition("Foundation")
+    end
+
+    psi_set = "poor (BC Hydro)"
+    io_path = ""
+    schema_path = File.dirname(__FILE__) + "/../tbd.schema.json"
+    gen_kiva = true
+    io, surfaces = processTBD(os_model, psi_set, io_path, schema_path, gen_kiva)
+    expect(surfaces.size).to eq(180)
+
+    kiva = false
+    surfaces.values.each do |surface|
+      next if kiva
+      kiva = true if surface.has_key?(:kiva)
+    end
+
+    expect(kiva).to be(true)
+    arg = "TotalExposedPerimeter"
+    foundation_kiva_settings = os_model.getFoundationKivaSettings
+    foundation_kiva_settings.setName("TBD-generated Kiva settings template")
+
+    # Generic 1" XPS insulation.
+    xps_25mm = OpenStudio::Model::StandardOpaqueMaterial.new(os_model)
```

Comment:
This is a Kiva question, but is this exterior insulation supposed to not be included in the Surface construction too?  Is there a way to check if it is (density < X?) and warn about possible double counting of insulation?

Reply:
No, this insulation material shouldn't be included as a construction layer elsewhere (although users can reuse the material in the future - as they see fit). This material is needed for the non-layered insulation configurations illustrated [here](https://github.com/rd2/tbd/blob/kiva/spec/files/KIVA.png).

I should nonetheless do a check first to ensure I'm not overwriting another existing material XPS material.